providers/openshift/provider.go

```diff
@@ -586,3 +580,18 @@ func request(client *http.Client, req *http.Request) (*simplejson.Json, error) {
 	}
 	return data, nil
 }
+
+func getKubeAPIURL() *url.URL {
```

Comment:
take `path string` as input to clean up some of the caller code.

Reply:
Done.